[2025-06-01T14:20:36.323+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: get_ingest_preprocess.get_data_from_api manual__2025-06-01T14:20:33.024338+00:00 [queued]>
[2025-06-01T14:20:36.340+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: get_ingest_preprocess.get_data_from_api manual__2025-06-01T14:20:33.024338+00:00 [queued]>
[2025-06-01T14:20:36.340+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-06-01T14:20:36.363+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): get_data_from_api> on 2025-06-01 14:20:33.024338+00:00
[2025-06-01T14:20:36.381+0000] {standard_task_runner.py:60} INFO - Started process 27929 to run task
[2025-06-01T14:20:36.396+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'get_ingest_preprocess', 'get_data_from_api', 'manual__2025-06-01T14:20:33.024338+00:00', '--job-id', '89', '--raw', '--subdir', 'DAGS_FOLDER/get_ingest_preprocess.py', '--cfg-path', '/tmp/tmp0j8tcmyy']
[2025-06-01T14:20:36.401+0000] {standard_task_runner.py:88} INFO - Job 89: Subtask get_data_from_api
[2025-06-01T14:20:36.521+0000] {task_command.py:423} INFO - Running <TaskInstance: get_ingest_preprocess.get_data_from_api manual__2025-06-01T14:20:33.024338+00:00 [running]> on host 9c020e596300
[2025-06-01T14:20:36.730+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='get_ingest_preprocess' AIRFLOW_CTX_TASK_ID='get_data_from_api' AIRFLOW_CTX_EXECUTION_DATE='2025-06-01T14:20:33.024338+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-01T14:20:33.024338+00:00'
[2025-06-01T14:20:36.735+0000] {logging_mixin.py:188} INFO - [INFO] Haciendo request a http://10.43.101.108:80/data con params {'group_number': 2, 'day': 'Tuesday'}
[2025-06-01T14:20:43.228+0000] {logging_mixin.py:188} INFO - [INFO] Primeras filas del batch recibido:
[2025-06-01T14:20:43.263+0000] {logging_mixin.py:188} INFO -    group_number      day  batch_number  ...  zip_code house_size  prev_sold_date
0             2  Tuesday             0  ...    6016.0     1617.0      1999-09-30
1             2  Tuesday             0  ...    6066.0     1850.0      2015-11-09
2             2  Tuesday             0  ...    6018.0     1620.0      2011-08-23
3             2  Tuesday             0  ...    6096.0     1735.0      2014-03-03
4             2  Tuesday             0  ...    6098.0     2005.0      2007-07-19

[5 rows x 15 columns]
[2025-06-01T14:20:44.883+0000] {logging_mixin.py:188} INFO - [ERROR] Fallo al guardar en la base de datos: (psycopg2.errors.UndefinedColumn) column "brokered_by" of relation "raw_data" does not exist
LINE 1: ...T INTO raw_data (group_number, day, batch_number, brokered_b...
                                                             ^

[SQL: INSERT INTO raw_data (group_number, day, batch_number, brokered_by, status, price, bed, bath, acre_lot, street, city, state, zip_code, house_size, prev_sold_date) VALUES (%(group_number)s, %(day)s, %(batch_number)s, %(brokered_by)s, %(status)s, %(price)s, %(bed)s, %(bath)s, %(acre_lot)s, %(street)s, %(city)s, %(state)s, %(zip_code)s, %(house_size)s, %(prev_sold_date)s)]
[parameters: ({'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 101640.0, 'status': 'for_sale', 'price': 289900.0, 'bed': 4.0, 'bath': 2.0, 'acre_lot': 0.38, 'street': 1758218.0, 'city': 'East Windsor', 'state': 'Connecticut', 'zip_code': 6016.0, 'house_size': 1617.0, 'prev_sold_date': '1999-09-30'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 107951.0, 'status': 'for_sale', 'price': 299900.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.87, 'street': 1336295.0, 'city': 'Vernon', 'state': 'Connecticut', 'zip_code': 6066.0, 'house_size': 1850.0, 'prev_sold_date': '2015-11-09'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 80935.0, 'status': 'for_sale', 'price': 299000.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.35, 'street': 920059.0, 'city': 'North Canaan', 'state': 'Connecticut', 'zip_code': 6018.0, 'house_size': 1620.0, 'prev_sold_date': '2011-08-23'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 33714.0, 'status': 'for_sale', 'price': 221000.0, 'bed': 4.0, 'bath': 2.0, 'acre_lot': 0.32, 'street': 731702.0, 'city': 'Windsor Locks', 'state': 'Connecticut', 'zip_code': 6096.0, 'house_size': 1735.0, 'prev_sold_date': '2014-03-03'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 29997.0, 'status': 'for_sale', 'price': 175000.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.19, 'street': 1382878.0, 'city': 'Winchester', 'state': 'Connecticut', 'zip_code': 6098.0, 'house_size': 2005.0, 'prev_sold_date': '2007-07-19'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 10571.0, 'status': 'for_sale', 'price': 140000.0, 'bed': 3.0, 'bath': 1.0, 'acre_lot': 0.21, 'street': 439965.0, 'city': 'Winchester', 'state': 'Connecticut', 'zip_code': 6098.0, 'house_size': 1595.0, 'prev_sold_date': '2007-10-17'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 51868.0, 'status': 'for_sale', 'price': 270000.0, 'bed': 5.0, 'bath': 3.0, 'acre_lot': 0.12, 'street': 1499905.0, 'city': 'Enfield', 'state': 'Connecticut', 'zip_code': 6082.0, 'house_size': 3117.0, 'prev_sold_date': '2014-07-01'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 33503.0, 'status': 'for_sale', 'price': 294900.0, 'bed': 5.0, 'bath': 3.0, 'acre_lot': 0.3, 'street': 1446997.0, 'city': 'Enfield', 'state': 'Connecticut', 'zip_code': 6082.0, 'house_size': 1772.0, 'prev_sold_date': '2005-08-15'}  ... displaying 10 of 73784 total bound parameter sets ...  {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 61013.0, 'status': 'for_sale', 'price': 300000.0, 'bed': 4.0, 'bath': 3.0, 'acre_lot': 1.9, 'street': 1031827.0, 'city': 'Fairbanks', 'state': 'Alaska', 'zip_code': 99709.0, 'house_size': 2304.0, 'prev_sold_date': '2015-04-27'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 21688.0, 'status': 'for_sale', 'price': 250000.0, 'bed': 4.0, 'bath': 3.0, 'acre_lot': 0.11, 'street': 1018367.0, 'city': 'Ketchikan', 'state': 'Alaska', 'zip_code': 99901.0, 'house_size': 2348.0, 'prev_sold_date': '2021-02-04'})]
(Background on this error at: https://sqlalche.me/e/14/f405)
[2025-06-01T14:20:44.884+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1890, in _execute_context
    self.dialect.do_executemany(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 982, in do_executemany
    context._psycopg2_fetched_rows = xtras.execute_values(
  File "/home/airflow/.local/lib/python3.8/site-packages/psycopg2/extras.py", line 1299, in execute_values
    cur.execute(b''.join(parts))
psycopg2.errors.UndefinedColumn: column "brokered_by" of relation "raw_data" does not exist
LINE 1: ...T INTO raw_data (group_number, day, batch_number, brokered_b...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/get_ingest_preprocess.py", line 61, in get_data_from_api
    raise e
  File "/opt/airflow/dags/get_ingest_preprocess.py", line 56, in get_data_from_api
    df.to_sql(RAW_TABLE, engine, if_exists="append", index=False)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 2878, in to_sql
    return sql.to_sql(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 769, in to_sql
    return pandas_sql.to_sql(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 1920, in to_sql
    total_inserted = sql_engine.insert_records(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 1470, in insert_records
    raise err
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 1461, in insert_records
    return table.insert(chunksize=chunksize, method=method)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 1023, in insert
    num_inserted = exec_insert(conn, keys, chunk_iter)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 929, in _execute_insert
    result = conn.execute(self.table.insert(), data)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1385, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1577, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1953, in _execute_context
    self._handle_dbapi_exception(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2134, in _handle_dbapi_exception
    util.raise_(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1890, in _execute_context
    self.dialect.do_executemany(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 982, in do_executemany
    context._psycopg2_fetched_rows = xtras.execute_values(
  File "/home/airflow/.local/lib/python3.8/site-packages/psycopg2/extras.py", line 1299, in execute_values
    cur.execute(b''.join(parts))
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column "brokered_by" of relation "raw_data" does not exist
LINE 1: ...T INTO raw_data (group_number, day, batch_number, brokered_b...
                                                             ^

[SQL: INSERT INTO raw_data (group_number, day, batch_number, brokered_by, status, price, bed, bath, acre_lot, street, city, state, zip_code, house_size, prev_sold_date) VALUES (%(group_number)s, %(day)s, %(batch_number)s, %(brokered_by)s, %(status)s, %(price)s, %(bed)s, %(bath)s, %(acre_lot)s, %(street)s, %(city)s, %(state)s, %(zip_code)s, %(house_size)s, %(prev_sold_date)s)]
[parameters: ({'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 101640.0, 'status': 'for_sale', 'price': 289900.0, 'bed': 4.0, 'bath': 2.0, 'acre_lot': 0.38, 'street': 1758218.0, 'city': 'East Windsor', 'state': 'Connecticut', 'zip_code': 6016.0, 'house_size': 1617.0, 'prev_sold_date': '1999-09-30'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 107951.0, 'status': 'for_sale', 'price': 299900.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.87, 'street': 1336295.0, 'city': 'Vernon', 'state': 'Connecticut', 'zip_code': 6066.0, 'house_size': 1850.0, 'prev_sold_date': '2015-11-09'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 80935.0, 'status': 'for_sale', 'price': 299000.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.35, 'street': 920059.0, 'city': 'North Canaan', 'state': 'Connecticut', 'zip_code': 6018.0, 'house_size': 1620.0, 'prev_sold_date': '2011-08-23'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 33714.0, 'status': 'for_sale', 'price': 221000.0, 'bed': 4.0, 'bath': 2.0, 'acre_lot': 0.32, 'street': 731702.0, 'city': 'Windsor Locks', 'state': 'Connecticut', 'zip_code': 6096.0, 'house_size': 1735.0, 'prev_sold_date': '2014-03-03'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 29997.0, 'status': 'for_sale', 'price': 175000.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.19, 'street': 1382878.0, 'city': 'Winchester', 'state': 'Connecticut', 'zip_code': 6098.0, 'house_size': 2005.0, 'prev_sold_date': '2007-07-19'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 10571.0, 'status': 'for_sale', 'price': 140000.0, 'bed': 3.0, 'bath': 1.0, 'acre_lot': 0.21, 'street': 439965.0, 'city': 'Winchester', 'state': 'Connecticut', 'zip_code': 6098.0, 'house_size': 1595.0, 'prev_sold_date': '2007-10-17'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 51868.0, 'status': 'for_sale', 'price': 270000.0, 'bed': 5.0, 'bath': 3.0, 'acre_lot': 0.12, 'street': 1499905.0, 'city': 'Enfield', 'state': 'Connecticut', 'zip_code': 6082.0, 'house_size': 3117.0, 'prev_sold_date': '2014-07-01'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 33503.0, 'status': 'for_sale', 'price': 294900.0, 'bed': 5.0, 'bath': 3.0, 'acre_lot': 0.3, 'street': 1446997.0, 'city': 'Enfield', 'state': 'Connecticut', 'zip_code': 6082.0, 'house_size': 1772.0, 'prev_sold_date': '2005-08-15'}  ... displaying 10 of 73784 total bound parameter sets ...  {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 61013.0, 'status': 'for_sale', 'price': 300000.0, 'bed': 4.0, 'bath': 3.0, 'acre_lot': 1.9, 'street': 1031827.0, 'city': 'Fairbanks', 'state': 'Alaska', 'zip_code': 99709.0, 'house_size': 2304.0, 'prev_sold_date': '2015-04-27'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 21688.0, 'status': 'for_sale', 'price': 250000.0, 'bed': 4.0, 'bath': 3.0, 'acre_lot': 0.11, 'street': 1018367.0, 'city': 'Ketchikan', 'state': 'Alaska', 'zip_code': 99901.0, 'house_size': 2348.0, 'prev_sold_date': '2021-02-04'})]
(Background on this error at: https://sqlalche.me/e/14/f405)
[2025-06-01T14:20:44.915+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=get_ingest_preprocess, task_id=get_data_from_api, execution_date=20250601T142033, start_date=20250601T142036, end_date=20250601T142044
[2025-06-01T14:20:44.937+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 89 for task get_data_from_api ((psycopg2.errors.UndefinedColumn) column "brokered_by" of relation "raw_data" does not exist
LINE 1: ...T INTO raw_data (group_number, day, batch_number, brokered_b...
                                                             ^

[SQL: INSERT INTO raw_data (group_number, day, batch_number, brokered_by, status, price, bed, bath, acre_lot, street, city, state, zip_code, house_size, prev_sold_date) VALUES (%(group_number)s, %(day)s, %(batch_number)s, %(brokered_by)s, %(status)s, %(price)s, %(bed)s, %(bath)s, %(acre_lot)s, %(street)s, %(city)s, %(state)s, %(zip_code)s, %(house_size)s, %(prev_sold_date)s)]
[parameters: ({'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 101640.0, 'status': 'for_sale', 'price': 289900.0, 'bed': 4.0, 'bath': 2.0, 'acre_lot': 0.38, 'street': 1758218.0, 'city': 'East Windsor', 'state': 'Connecticut', 'zip_code': 6016.0, 'house_size': 1617.0, 'prev_sold_date': '1999-09-30'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 107951.0, 'status': 'for_sale', 'price': 299900.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.87, 'street': 1336295.0, 'city': 'Vernon', 'state': 'Connecticut', 'zip_code': 6066.0, 'house_size': 1850.0, 'prev_sold_date': '2015-11-09'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 80935.0, 'status': 'for_sale', 'price': 299000.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.35, 'street': 920059.0, 'city': 'North Canaan', 'state': 'Connecticut', 'zip_code': 6018.0, 'house_size': 1620.0, 'prev_sold_date': '2011-08-23'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 33714.0, 'status': 'for_sale', 'price': 221000.0, 'bed': 4.0, 'bath': 2.0, 'acre_lot': 0.32, 'street': 731702.0, 'city': 'Windsor Locks', 'state': 'Connecticut', 'zip_code': 6096.0, 'house_size': 1735.0, 'prev_sold_date': '2014-03-03'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 29997.0, 'status': 'for_sale', 'price': 175000.0, 'bed': 3.0, 'bath': 2.0, 'acre_lot': 0.19, 'street': 1382878.0, 'city': 'Winchester', 'state': 'Connecticut', 'zip_code': 6098.0, 'house_size': 2005.0, 'prev_sold_date': '2007-07-19'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 10571.0, 'status': 'for_sale', 'price': 140000.0, 'bed': 3.0, 'bath': 1.0, 'acre_lot': 0.21, 'street': 439965.0, 'city': 'Winchester', 'state': 'Connecticut', 'zip_code': 6098.0, 'house_size': 1595.0, 'prev_sold_date': '2007-10-17'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 51868.0, 'status': 'for_sale', 'price': 270000.0, 'bed': 5.0, 'bath': 3.0, 'acre_lot': 0.12, 'street': 1499905.0, 'city': 'Enfield', 'state': 'Connecticut', 'zip_code': 6082.0, 'house_size': 3117.0, 'prev_sold_date': '2014-07-01'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 33503.0, 'status': 'for_sale', 'price': 294900.0, 'bed': 5.0, 'bath': 3.0, 'acre_lot': 0.3, 'street': 1446997.0, 'city': 'Enfield', 'state': 'Connecticut', 'zip_code': 6082.0, 'house_size': 1772.0, 'prev_sold_date': '2005-08-15'}  ... displaying 10 of 73784 total bound parameter sets ...  {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 61013.0, 'status': 'for_sale', 'price': 300000.0, 'bed': 4.0, 'bath': 3.0, 'acre_lot': 1.9, 'street': 1031827.0, 'city': 'Fairbanks', 'state': 'Alaska', 'zip_code': 99709.0, 'house_size': 2304.0, 'prev_sold_date': '2015-04-27'}, {'group_number': 2, 'day': 'Tuesday', 'batch_number': 1, 'brokered_by': 21688.0, 'status': 'for_sale', 'price': 250000.0, 'bed': 4.0, 'bath': 3.0, 'acre_lot': 0.11, 'street': 1018367.0, 'city': 'Ketchikan', 'state': 'Alaska', 'zip_code': 99901.0, 'house_size': 2348.0, 'prev_sold_date': '2021-02-04'})]
(Background on this error at: https://sqlalche.me/e/14/f405); 27929)
[2025-06-01T14:20:45.077+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-06-01T14:20:45.113+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
